### PR TITLE
GH-84

### PR DESCRIPTION
**Pull request by evdhoudt**
Closes NCI-Agency/anet#84

### CODE REVIEW

**Comment by evdhoudt** [comment]
I'm not really sure why the build is failing.. anybody ideas what I have to fix?

File: insertBaseData.sql [hunk 1]
```diff
@@ -580,6 +580,7 @@ INSERT INTO adminSettings ([key], value) VALUES ('HELP_LINK_URL', 'http://google
 INSERT INTO adminSettings ([key], value) VALUES ('CONTACT_EMAIL', 'team-anet@dds.mil');
 INSERT INTO adminsettings ([key], value) VALUES ('DAILY_ROLLUP_MAX_REPORT_AGE_DAYS', '14');
 INSERT INTO adminsettings ([key], value) VALUES ('EXTERNAL_DOCUMENTATION_LINK_URL', '');
+<<<<<<< HEAD
```
Comment (VassilIordanov) [inline insertBaseData.sql]: Doesn't this need to be merged?

**Comment by VassilIordanov** [comment]
@evdhoudt build fails because of insertBaseData.sql, see my comment in the file

File: src/main/resources/migrations.xml [hunk 2]
```diff
@@ -686,6 +686,40 @@
...
+			</delete>
+		</rollback>
+	</changeSet>
+
+	<changeSet id="add-banner-visibility" author="evdhoudt">
+		<insert tableName="adminSettings">
+			<column name="key" value="GENERAL_BANNER_VISIBILITY" />
+			<column name="value" value="1" />
+		</insert>
+		<rollback>
```
Comment (VassilIordanov) [inline src/main/resources/migrations.xml]: missing closing tag

File: src/main/resources/migrations.xml [hunk 3]
```diff
@@ -647,7 +647,7 @@
 	</changeSet>
 
 	<changeSet id="35" author="evdhoudt" dbms="mssql" >
-		<sql dbms="mssql">
+		<sql dbms="">
```
Comment (maradragan) [inline src/main/resources/migrations.xml]: I would either keep dbms="mssql" or just remove the dbms here.

File: src/main/resources/migrations.xml (hunk 2 above)
Comment (evdhoudt) [inline src/main/resources/migrations.xml]: fixed

File: src/main/resources/migrations.xml (hunk 3 above)
Comment (evdhoudt) [inline src/main/resources/migrations.xml]: restored to original settings dbms="mssql"

File: client/src/components/GeneralBanner.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    const banner = props.options
+    if (banner.visible && banner.message){
+        return (
+            <div className={bannerClassName(banner.level)}>
+                <div className="messsage"><strong>{banner.title}:</strong> {banner.message}</div>
+            </div>
+        )
+    } else { return null }
+}
+export default GeneralBanner
```
Comment (maradragan) [inline client/src/components/GeneralBanner.js]: No newline at end of file

File: client/src/components/TopBar.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+    componentDidUpdate() {
+        this.updateBodyPaddingTop()
+        this.updateBannerVisibility()
+    }
+
+    updateBodyPaddingTop() {
+        let topbarPaddingHeight = document.getElementById('topbar').offsetHeight + 20
+        if (this.state.bodyPaddingTop !== topbarPaddingHeight){
+            document.body.style.paddingTop=`${topbarPaddingHeight}px`
```
Comment (maradragan) [inline client/src/components/TopBar.js]: Empty space missing around the =.

File: client/src/components/GeneralBanner.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+
+function bannerClassName(level) {
+    var output = 'general-banner alert'
```
Comment (maradragan) [inline client/src/components/GeneralBanner.js]: I would use here also the notice level as default.

File: docs/User_Workflows_and_Documentation.md [hunk 7]
```diff
@@ -254,3 +254,22 @@ To set the default approval organization, start by selecting “Admins” from t
 **Updating the Security Banner Text and Color**
 
 Admins have the ability to modify the color and text of the security banner. To modify these settings, start by clicking on “Admins” on the left-hand menu on the home screen. From there, you can change the text of the security banner by entering your desired text in the “Security banner text” field. Similarly, you can change the color by entering your desired color in the “Security banner color” field. 
+
+**Updating the General Banner Text and Announcement Level and banner visibility**
+
+Admins have the ability to modify the color and text of the general anouncement banner. To modify these settings, start by clicking on “Admins” on the left-hand menu on the home screen.
+
+From there, you can change the text of the general banner by entering your desired text in the General banner text” field.
```
Comment (maradragan) [inline docs/User_Workflows_and_Documentation.md]: Missing start quotes.

**Pull request merged**